backend/src/main/kotlin/org/icpclive/service/QueueService.kt

```diff
@@ -69,21 +82,40 @@ class QueueService {
                 }
                 is Run -> {
                     val run = event.run
-                    val currentTime = contestInfoFlow.value.currentContestTime.takeIf { it != firstEventTime } ?: run.time
+                    val currentTime =
+                        contestInfoFlow.value.currentContestTime.takeIf { it != firstEventTime } ?: run.time
                     logger.debug("Receive run $run")
                     lastUpdateTime[run.id] = currentTime
                     if (run.id in runs || contestInfoFlow.value.currentContestTime <= currentTime + run.timeInQueue) {
-                        resultFlow.emit(if (run.id in runs) ModifyRunInQueueEvent(run) else AddRunToQueueEvent(run))
-                        runs[run.id] = run
+                        modifyRun(run)
```

Comment:
I think we need to preserve isFeaturedRun here. As now update of featured run will make it unfeatured